### ISSUE TITLE
iPython kernel fails to start due to space in directory name

### ISSUE THREAD

**Issue by fishter** [spyder-ide/spyder#21909]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [X] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X] Tried basic troubleshooting (if a bug/error)
    * [X] Restarted Spyder
    * [X] Reset preferences with ``spyder --reset``
    * [X] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X] Tried the other applicable steps from the Troubleshooting Guide
* [X] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
When Spyder is started there is an error message in the console. The error message states that Spyder could not start the python kernel.
The error message implies that the problem is due to a space in the user profile path. Spaces in paths are not an uncommon feature of modern filesystems. Spyder should be able to cope with a space in a directory or filename.

### What steps reproduce the problem?

1. On Windows, install Spyder (via Anaconda, or direct) for a user with a space in their profile name.
2. Start Spyder.

### What is the expected output? What do you see instead?
Expected output is a usable console which has connected to a running python kernel. The actual outcome is an error message stating that a connection could not be made.

### Error 
from iPython Console

```
An error ocurred while starting the kernel
'C:/Users/Graeme' is not recognized as an internal or external command,
operable program or batch file.
'C:\Users\Graeme' is not recognized as an internal or external command,
operable program or batch file.
```

running ```jupyter qtconsole``` is OK.

```
Jupyter QtConsole 5.5.1
Python 3.9.18 | packaged by conda-forge | (main, Aug 30 2023, 03:40:31) [MSC v.1929 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: pwd
Out[1]: 'C:\\Users\\Graeme Hilton local'

In [2]: 
```
Connecting to a remote kernel is OK.

## Versions
* Spyder version: 5.5.3  (conda)
* Python version: 3.9.18 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.10
* Operating System: Windows-10-10.0.22631-SP0

### Dependencies
```
# Mandatory:
atomicwrites >=1.2.0             :  1.4.1 (OK)
chardet >=2.0.0                  :  5.2.0 (OK)
cloudpickle >=0.5.0              :  3.0.0 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20230430 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.15.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.19.1 (OK)
jellyfish >=0.7                  :  1.0.3 (OK)
jsonschema >=3.2.0               :  4.21.1 (OK)
keyring >=17.0.0                 :  24.3.1 (OK)
nbconvert >=4.0                  :  None (NOK)
numpydoc >=0.6.0                 :  1.6.0 (OK)
paramiko >=2.4.0                 :  3.4.0 (OK)
parso >=0.7.0,<0.9.0             :  0.8.3 (OK)
pexpect >=4.4.0                  :  4.9.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  5.9.8 (OK)
pygments >=2.0                   :  2.17.2 (OK)
pylint >=2.5.0,<3.1              :  2.16.2 (OK)
pylint_venv >=3.0.2              :  3.0.3 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.10.0,<1.11.0           :  1.10.1 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.2 (OK)
qtawesome >=1.2.1,<1.3.0         :  1.2.3 (OK)
qtconsole >=5.5.1,<5.6.0         :  5.5.1 (OK)
qtpy >=2.1.0                     :  2.4.1 (OK)
rtree >=0.9.7                    :  1.2.0 (OK)
setuptools >=49.6.0              :  69.2.0 (OK)
sphinx >=0.6.6                   :  5.0.2 (OK)
spyder_kernels >=2.5.1,<2.6.0    :  2.5.1 (OK)
textdistance >=4.2.0             :  4.5.0 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  4.0.0 (OK)
zmq >=24.0.0                     :  25.1.2 (OK)

# Optional:
cython >=0.21                    :  None (NOK)
matplotlib >=3.0.0               :  3.8.3 (OK)
numpy >=1.7                      :  1.26.4 (OK)
pandas >=1.1.1                   :  2.2.1 (OK)
scipy >=0.17.0                   :  1.12.0 (OK)
sympy >=0.7.3                    :  1.12 (OK)
```

**Comment by LeeEtchels** [comment]
I'm having a similar issue with an underscore instead of a space in my user dir. It shows up as a permission error, permission denied to the incorrect user dir which I am assuming is because it doesn't exist. Just mentioning in case it is related and helps to narrow it down while I search for a solution.

**Comment by ccordoba12** [comment]
Hey folks, thanks for reporting. This is duplicate of issue #21662 and we'll try to fix it in our next version (5.5.4), to be released next week.
